plugins/woocommerce/client/admin/client/blueprint/components/BlueprintUploadDropzone.tsx

```diff
@@ -91,17 +91,42 @@ const importBlueprint = async ( file: File ) => {
 			throw new Error( 'Invalid JSON format: Expected an array.' );
 		}
 
+		const MAX_STEP_SIZE_BYTES =
+			window?.wcSettings?.admin?.blueprint_max_step_size_bytes ||
+			50 * 1024 * 1024; // defaults to 50MB
+
 		// Loop through each step and send it to the endpoint
 		for ( const step of steps ) {
+			const stepJson = JSON.stringify( {
+				step_definition: step,
+			} );
+			const stepSize = new Blob( [ stepJson ] ).size;
+			if ( stepSize > MAX_STEP_SIZE_BYTES ) {
```

Comment:
This is all working really well and the file I created using @rjchow's script is firing off this error. 

Am I reading correctly that this is just measuring the step's size, not the entire file's size? If so, wouldn't that allow file sizes larger than the limit? I also ask because I see the script places all the dummy data into a single step.

Reply:
@psealock Thank you!

> Am I reading correctly that this is just measuring the step's size, not the entire file's size? If so, wouldn't that allow file sizes larger than the limit? I also ask because I see the script places all the dummy data into a single step.

You're correct. We load the file in the browser and only upload each step to the server to process. I think it's okay to allow a large file as we don't upload it to the server. I don't have a strong preference though. Do you think we should limit the total filesize instead?

Reply:
Gotcha, that totally makes sense!